common/tmx_filetype.go

```diff
@@ -0,0 +1,60 @@
+package common
+
+import (
+	"bytes"
+	"fmt"
+        "engo.io/engo"
+	"io"
+)
+
+// TMXResource contains a level created from a Tile Map XML
+type TMXResource struct {
+	Level *Level
+	url   string
+}
+
+func (r TMXResource) URL() string {
+	return r.url
+}
+
+// tmxLoader is responsible for managing '.tmx' files within 'engo.Files'
+type tmxLoader struct {
+	levels map[string]TMXResource
+}
+
+// Load will load the tmx file and any other image resources that are needed
```

Comment:
I really don't think this is necessary... googling tmx shows tiled map editor, it's the tiled map editor format, no one else uses it.